### PR TITLE
[Doc] Performance tuning update

### PR DESCRIPTION
**Pull request by qiao-bo**
Related issue = #

### CODE REVIEW

**Review by k-ye — APPROVED** [review]
LGTM! Could you also add something to mention that layout could also impact the performance, e.g., https://docs.taichi.graphics/docs/lang/articles/advanced/layout#array-of-structures-aos-structure-of-arrays-soa ?

File: docs/lang/articles/advanced/performance.md [hunk 1]
```diff
@@ -6,33 +6,29 @@ sidebar_position: 5
...
 
-From small to large, the computation units are: **iteration** \<
-**thread** \< **block** \< **grid**.
+From a fine-grained to a coarse-grained level, the computation units can be
+defined as: **iteration** \< **thread** \< **block** \< **grid**.
 
 - **iteration**: Iteration is the **body of a for-loop**. Each
   iteration corresponding to a specific `i` value in for-loop.
-- **thread**: Iterations are grouped into threads. Threads are the
+- **thread**: Iterations are grouped into threads. Thread is the
```
Comment (erizmr) [inline docs/lang/articles/advanced/performance.md:31]:
```suggestion
- **thread**: Iterations are grouped into threads. A thread is the
```

File: docs/lang/articles/advanced/performance.md [hunk 2]
```diff
@@ -6,33 +6,29 @@ sidebar_position: 5
...
-GPUs have a **thread hierarchy**.
+To better understand how the mentioned for-loop is parallelized, we briefly
+introduce the **thread hierarchy** on modern GPU architectures.
 
-From small to large, the computation units are: **iteration** \<
-**thread** \< **block** \< **grid**.
+From a fine-grained to a coarse-grained level, the computation units can be
+defined as: **iteration** \< **thread** \< **block** \< **grid**.
 
 - **iteration**: Iteration is the **body of a for-loop**. Each
```
Comment (erizmr) [inline docs/lang/articles/advanced/performance.md:29]:
```suggestion
- **iteration**: An iteration is the **body of a for-loop**. Each
```

**Review by erizmr — COMMENTED** [review]
LGTM! Just a few nits.

**Review by erizmr — APPROVED** [review]

**Pull request merged**